src/thunderbots/software/backend_output/grsim/motion_controller.cpp

```diff
@@ -0,0 +1,206 @@
+                                                 const double delta_time)
+{
+    // vector to hold the XY velocities of the robot
+    Vector robot_linear_velocities;
+
+    // boolean value if the robot can reach it's destination at target speed based on
+    // linear MAX acceleration
+    bool can_stop_in_time;
+
+    // destination distance used for constant linear acceleration speed calculations
+    const double distance_to_dest = (robot.position() - dest).len();
+
+    // calculates robot angle based on unit vector that points from the robot location to
+    // the destination (used to calculate the X/Y velocity magnitudes
+    const Angle direction_angle = (dest - robot.position()).norm().orientation();
+    double expected_final_speed;
+
+    double delta_speed_x, delta_speed_y;
+
+    // calculate the expected speed at the destination based on current speed and
+    // acceleration Vf = sqrt( Vi^2 - 2*a*d)
+
+    // check for negative sqrt case
+    if (pow(robot.velocity().len(), 2) <= 2 * ROBOT_MAX_ACCELERATION * distance_to_dest)
+    {
+        // if the sqrt is negative, the final speed will be negative (opposite direction
+        // of current speed)
+        expected_final_speed = -1 * sqrt(2 * ROBOT_MAX_ACCELERATION * distance_to_dest -
+                                         pow(robot.velocity().len(), 2));
+        can_stop_in_time     = true;
+    }
+    else
+    {
+        // calculate the expected final speed assuming max decceleration
+        expected_final_speed = sqrt(pow(robot.velocity().len(), 2) -
+                                    2 * ROBOT_MAX_ACCELERATION * distance_to_dest);
+
+        // the robot can stop in time if it's desired final speed is higher than the speed
+        // if the robot maximum decelerates from the current state
+        can_stop_in_time = expected_final_speed <= desired_final_speed;
+    }
+
+    if (can_stop_in_time)
+    {
+        // if the robot can stop in time and is going slower than the desired final speed
+        // then accelerate
+        if (expected_final_speed < desired_final_speed)
+        {
+            delta_speed_x = (ROBOT_MAX_ACCELERATION * delta_time) * direction_angle.cos();
+            delta_speed_y = (ROBOT_MAX_ACCELERATION * delta_time) * direction_angle.sin();
+        }
+
+        // if not then maintain speed
+        else
+        {
+            // maintain speed
+            delta_speed_x = 0.0;
+            delta_speed_y = 0.0;
+        }
+    }
+
+    // if the robot can't stop in time then decelerate
+    else
+    {
+        delta_speed_x = -(ROBOT_MAX_ACCELERATION * delta_time) * direction_angle.cos();
+        delta_speed_y = -(ROBOT_MAX_ACCELERATION * delta_time) * direction_angle.sin();
+    }
+
+    robot_linear_velocities =
+        Vector(robot.velocity().x() + delta_speed_x,
+               robot.velocity().y() +
+                   delta_speed_y);  // calculate new X/Y/ang velocities based on the
```

Comment:
nit: move the comment to line 193, and move `delta_speed_y);` one line up

Reply:
Formatted this statement better.